ethclient/ethclient.go

```diff
@@ -339,10 +339,77 @@ func (ec *Client) GetBlockTraceByNumber(ctx context.Context, number *big.Int) (*
 
 // GetBlockByHash returns the requested block. When fullTx is true all transactions in the block are returned in full
 // detail, otherwise only the transaction hash is returned.
-func (ec *Client) GetBlockByHash(ctx context.Context, blockHash common.Hash, inclTx bool) (map[string]interface{}, error) {
-	block := make(map[string]interface{})
-	err := ec.c.CallContext(ctx, &block, "scroll_getBlockByHash", blockHash, inclTx)
-	return block, err
+func (ec *Client) GetBlockByHash(ctx context.Context, blockHash common.Hash) (*types.BlockWithRowConsumption, error) {
+	var raw json.RawMessage
+	err := ec.c.CallContext(ctx, &raw, "scroll_getBlockByHash", blockHash, true)
+	if err != nil {
+		return nil, err
+	} else if len(raw) == 0 {
+		return nil, ethereum.NotFound
+	}
+	// Decode header and transactions.
+	var head *types.Header
+	var body rpcBlock
+	var rc *types.RowConsumption
+	if err := json.Unmarshal(raw, &head); err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(raw, &body); err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(raw, &rc); err != nil {
+		rc = nil
```

Comment:
Can add a comment here, e.g., if the node doesn't have this data, it will return `nil`.

Reply:
added a comment